plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/metadata/MetadataUpdater.java

```diff
@@ -0,0 +1,237 @@
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+import org.sonatype.nexus.repository.maven.MavenFacet;
+import org.sonatype.nexus.repository.maven.MavenPath;
+import org.sonatype.nexus.repository.maven.MavenPath.HashType;
+import org.sonatype.nexus.repository.maven.internal.maven2.Maven2Format;
+import org.sonatype.nexus.repository.maven.internal.maven2.Maven2MetadataMerger;
+import org.sonatype.nexus.repository.maven.internal.maven2.Maven2MetadataMerger.MetadataEnvelope;
+import org.sonatype.nexus.repository.maven.internal.maven2.metadata.Maven2Metadata.Plugin;
+import org.sonatype.nexus.repository.maven.internal.maven2.metadata.Maven2Metadata.Snapshot;
+import org.sonatype.nexus.repository.util.TypeTokens;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.payloads.BytesPayload;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.hash.HashCode;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.artifact.repository.metadata.SnapshotVersion;
+import org.apache.maven.artifact.repository.metadata.Versioning;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Writer;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Maven 2 repository metadata updater.
+ *
+ * @since 3.0
+ */
+public class MetadataUpdater
+    extends ComponentSupport
+{
+  private final MavenFacet mavenFacet;
+
+  private final Maven2MetadataMerger metadataMerger;
+
+  private final MetadataXpp3Reader metadataReader;
+
+  private final MetadataXpp3Writer metadataWriter;
+
+  private final DateFormat dotlessTimestampFormat;
+
+  private final DateFormat dottedTimestampFormat;
+
+  @Inject
+  public MetadataUpdater(final MavenFacet mavenFacet) {
+    this.mavenFacet = checkNotNull(mavenFacet);
+    this.metadataMerger = new Maven2MetadataMerger();
+    this.metadataReader = new MetadataXpp3Reader();
+    this.metadataWriter = new MetadataXpp3Writer();
+    this.dotlessTimestampFormat = new SimpleDateFormat("yyyyMMddHHmmss");
+    this.dotlessTimestampFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    this.dottedTimestampFormat = new SimpleDateFormat("yyyyMMdd.HHmmss");
```

Comment:
To be precise, dotless is once (here), and dotted is twice (here and Maven2MavenPathParser) defined :) But agree, ack, fixing.